hibernate-vector/src/main/java/org/hibernate/vector/AbstractOracleVectorJdbcType.java

```diff
@@ -0,0 +1,177 @@
+		);
+	}
+
+	@Override
+	public String toString() {
+		return "OracleVectorTypeDescriptor";
+	}
+
+	/**
+	 * Checks if the Oracle JDBC driver version is less than 23.4.
+	 *
+	 * @return true if the driver version is less than 23.4, false otherwise
+	 */
+	protected boolean isVectorSupportedByDriver(WrapperOptions options) throws SQLException {
+		OracleDialect dialect = (OracleDialect) options.getSessionFactory().getJdbcServices().getDialect();
+
+		int majorVersion = dialect.getDriverMajorVersion();
+		int minorVersion = dialect.getDriverMinorVersion();
+
+		return ( majorVersion > 23 ) || ( majorVersion == 23 && minorVersion >= 4 );
+	}
+
+	@Override
+	public <X> ValueBinder<X> getBinder(final JavaType<X> javaTypeDescriptor) {
+		return new BasicBinder<>( javaTypeDescriptor, this ) {
+			@Override
+			protected void doBind(PreparedStatement st, X value, int index, WrapperOptions options)
+					throws SQLException {
+				if ( isVectorSupportedByDriver(options) ) {
+					st.setObject( index, value, OracleType.valueOf( getFriendlyName() ) );
+				}
+				else {
+					st.setString( index, getStringVector( value, javaTypeDescriptor, options ) );
+				}
+			}
+
+			@Override
+			protected void doBind(CallableStatement st, X value, String name, WrapperOptions options)
+					throws SQLException {
+				if ( isVectorSupportedByDriver(options) ) {
+					st.setObject( name, value, OracleType.valueOf( getFriendlyName() ) );
+				}
+				else {
+					st.setString( name, getStringVector( value, javaTypeDescriptor, options ) );
+				}
+			}
+
+
+		};
+	}
+
+	@Override
+	public <X> ValueExtractor<X> getExtractor(final JavaType<X> javaTypeDescriptor) {
+
+		return new BasicExtractor<>( javaTypeDescriptor, this ) {
+			@Override
+			protected X doExtract(ResultSet rs, int paramIndex, WrapperOptions options) throws SQLException {
+				if ( isVectorSupportedByDriver(options) ) {
+					return javaTypeDescriptor.wrap( rs.getObject( paramIndex, javaType ), options );
+				}
+				else {
+					return javaTypeDescriptor.wrap( getVectorArray( rs.getString( paramIndex ) ), options );
+				}
+			}
+
+			@Override
+			protected X doExtract(CallableStatement statement, int index, WrapperOptions options) throws SQLException {
+				if ( isVectorSupportedByDriver(options) ) {
+					return javaTypeDescriptor.wrap( statement.getObject( index, javaType ), options );
+				}
+				else {
+					return javaTypeDescriptor.wrap( getVectorArray( statement.getString( index ) ), options );
+				}
+			}
+
+			@Override
+			protected X doExtract(CallableStatement statement, String name, WrapperOptions options)
+					throws SQLException {
+				if ( isVectorSupportedByDriver(options) ) {
+					return javaTypeDescriptor.wrap( statement.getObject( name, javaType ), options );
+				}
+				else {
+					return javaTypeDescriptor.wrap( getVectorArray( statement.getString( name ) ), options );
+				}
+			}
+
+
+		};
+	}
+
+	protected abstract <T> T getVectorArray(String string);
+
```

Comment:
Please add constants to `org.hibernate.dialect.OracleTypes`:

```suggestion

	protected abstract Class<?> getNativeTypeCode();
```